utils.js

```diff
@@ -7243,3 +7107,195 @@ export const parseCUsageSlice = (sliceData) => {
+    }
+    const firstItem = items[0];
+    // Work backwards to find the body for the matching version
+    body.concat(firstItem.items[firstItem.items.length - 1])
+    if (np.version) {
+      newBody.push(firstItem.items
+        .reverse()
+        .filter(
+          (i) => i.catalogEntry && i.catalogEntry.version === np.version
+        ));
+    }
+  }
+  return [np, newBody, body]
+}
+
+/**
+ * Method to retrieve metadata for nuget packages
+ *
+ * @param {Array} pkgList Package list
+ */
+export const getNugetMetadata = async function (pkgList, dependencies = undefined) {
+  const NUGET_URL = await getNugetUrl();
+  const cdepList = [];
+  const depRepList = {};
+  for (const p of pkgList) {
+    let cacheKey = undefined;
+    try {
+      // If there is a version, we can safely use the cache to retrieve the license
+      // See: https://github.com/CycloneDX/cdxgen/issues/352
+      cacheKey = `${p.group}|${p.name}`;
+      let body = metadata_cache[cacheKey];
+
+      if (body && body.error) {
+        cdepList.push(p);
+        continue;
+      }
+      if (!body) {
+        let newBody = {};
+        let np = {};
+        [np, newBody, body] = await queryNuget(p, NUGET_URL)
+        if (p.version !== np.version) {
+          const oldRef = decodeURIComponent(
+              new PackageURL(
+                "nuget",
+                "",
+                p.name,
+                p.version,
+                null,
+                null
+              ).toString()
+            );
+          depRepList[oldRef] = decodeURIComponent(
+            new PackageURL(
+              "nuget",
+              "",
+              np.name,
+              np.version,
+              null,
+              null
+            ).toString()
+          );
+          p.version = np.version;
+        }
+        if (newBody && newBody[0].length > 0) {
+          body = newBody[0][0];
+        }
+        if (body) {
+          metadata_cache[cacheKey] = body;
+          // Set the latest version in case it is missing
+          if (!p.version && body.catalogEntry.version) {
+            p.version = body.catalogEntry.version;
+          }
+          p.description = body.catalogEntry.description;
+          if (body.catalogEntry.authors) {
+            p.author = body.catalogEntry.authors.trim();
+          }
+          if (
+            body.catalogEntry.licenseExpression &&
+            body.catalogEntry.licenseExpression !== ""
+          ) {
+            p.license = findLicenseId(body.catalogEntry.licenseExpression);
+          } else if (body.catalogEntry.licenseUrl) {
+            p.license = findLicenseId(body.catalogEntry.licenseUrl);
+          }
+          if (body.catalogEntry.projectUrl) {
+            p.repository = {url: body.catalogEntry.projectUrl};
+            p.homepage = {
+              url:
+                "https://www.nuget.org/packages/" +
+                p.group +
```

Comment:
Can we remove p.group from the url completely now that we have ensured only the name field is used through out?